docs-internal/db.md

```diff
@@ -0,0 +1,364 @@
+- `CompletionStage<Void> consume(Consumer<DbRowResult<T>> consumer)`: allows to register consumer of database rows
+- `CompletionStage<List<T>> collect()`: allows to retrieve and process result as List of database rows. This method is limited to small result sets.
+
+### Database Row
+
+Database rows returned by query statement execution are returned as `DbRow` interface. This interface allows direct column access. It's also possible to map this row to another class instance using `DbMapper<T>` interface.
+
+## Transactions
+
+Transactions are supported using similar code pattern like simple statement execution. The only difference is usage of `<T> T inTransaction(Function<HelidonDbExecute, T> executor)` method instead of `execute`. Transaction scope and lifecycle is bound to `executor` instance.
+Transaction is committed at the end of `executor` function scope when no exception has been thrown. Any exception thrown from `executor` function will cause transaction rollback.
+
+## Example
+
+### Application Configuration File
+A configuration file used when creating a `Config` instance:
+```yaml
+helidon-db:
+    source: jdbc
+    url: jdbc:mysql://127.0.0.1:3306/myDatabase
+    username: user
+    password: password
+    statements:
+        insert-indexed-params: "INSERT INTO my_table VALUES(?, ?)"
+        insert-named-params:   "INSERT INTO my_table VALUES(:name, :type)"
+        select-all: "SELECT * FROM my_table"
+        select-indexed-params: "SELECT * FROM my_table WHERE name = ?"
+        select-named-params:   "SELECT * FROM my_table WHERE name = :name"
+```
+
+### Database Initialization:
+```java
+HelidonDb db = config
+                .get("helidon-db")
+                .as(HelidonDb::create)
+                .orElseThrow(() -> new IllegalStateException("Configuration is missing"));
+```
+
+### Statements Execution:
+
+#### Insert with Indexed Parameters
+```java
+db.execute(exec -> exec
+                .createNamedDmlStatement("insert-indexed-params")
+                .addParam("Pikachu")
+                .addParam("electric")
+                .execute())
+          .thenAccept(count -> <do something>)
+          .exceptionally(throwable -> <fail somehow>);
+```
+
+#### Insert with Named Parameters
+```java
+db.execute(exec -> exec
+                .createNamedDmlStatement("insert-named-params")
+                .addParam("name", "Pikachu")
+                .addParam("type", "electric")
+                .execute()
+          )
+          .thenAccept(count -> <do something>)
+          .exceptionally(throwable -> <fail somehow>);
+```
+
+#### Query without parameters
+
+```java
+db.execute(exec -> exec.namedQuery("select-all"))
+                .consume(response::send)
+                .exceptionally(throwable -> sendError(throwable, response));
+```
+
+#### Get with indexed parameters
+The `get` method is a shortcut to a query method that expects zero to 1 results.
```

Comment:
I guess this is `get` or `namedGet` as below?

Reply:
I meant a general `get` approach - there are four methods (`get`, `namedGet`, `createGet`, and `createNamedGet`.

Reply:
full list provided